plutus-metatheory/src/Algorithmic/CEK.lagda.md

```diff
@@ -72,43 +68,47 @@ data Value : (A : ∅ ⊢Nf⋆ *) → Set where
    → Value (nf (embNf A · ƛ (μ (embNf (weakenNf A)) (` Z)) · embNf B))
    → Value (μ A B)
 
-  V-con : {tcn : TyCon _}
-    → (cn : TermCon {∅} (con tcn))
-    → Value (con tcn)
+  V-con : ∀{A : ∅ ⊢Nf⋆ ♯}
+    → ⟦ A ⟧ 
+    → Value (con A)
 
   V-I⇒ : ∀ b {A B}{tn}
-       → {pt : tn ∔ 0 ≣ fv♯ (signature b)} 
+       → {pt : tn ∔ 0 ≣ fv (signature b)} 
        → ∀{an am}{pa : an ∔ (suc am) ≣ args♯ (signature b)}
        → {σB : SigTy pt (bubble pa) B}
        → BApp b (A ⇒ B) (A B⇒ σB)
        → Value (A ⇒ B)
 
-  V-IΠ : ∀ b {B : ∅ ,⋆ * ⊢Nf⋆ *}
-       → ∀{tn tm} {pt : tn ∔ (suc tm) ≣ fv♯ (signature b)} 
+  V-IΠ : ∀ b {K}{B : ∅ ,⋆ K ⊢Nf⋆ *}
+       → ∀{tn tm} {pt : tn ∔ (suc tm) ≣ fv (signature b)} 
        → ∀{an am}{pa : an ∔ suc am ≣ args♯ (signature b)}
        → {σB : SigTy (bubble pt) pa B}
        → BApp b (Π B) (sucΠ σB)
        → Value (Π B)
 
 data BApp b where
   base : BApp b (sig2type (signature b)) (sig2SigTy (signature b))
-  app : ∀{A B}{tn}
-    → {pt : tn ∔ 0 ≣ fv♯ (signature b)} 
+  _$_ : ∀{A B}{tn}
+    → {pt : tn ∔ 0 ≣ fv (signature b)} 
     → ∀{an am}{pa : an ∔ suc am ≣ args♯ (signature b)}
     → {σB : SigTy pt (bubble pa) B}
     → BApp b (A ⇒ B) (A B⇒ σB)
     → Value A 
     → BApp b B σB
-  app⋆ : ∀{B C}
-    → ∀{tn tm} {pt : tn ∔ (suc tm) ≣ fv♯ (signature b)} 
+  _$$_ : ∀{K}{B : ∅ ,⋆ K ⊢Nf⋆ * }{C}
+    → ∀{tn tm} {pt : tn ∔ (suc tm) ≣ fv (signature b)} 
     → ∀{an am}{pa : an ∔ (suc am) ≣ args♯ (signature b)}
     → {σB : SigTy (bubble pt) pa B}
     → BApp b (Π B) (sucΠ σB)
-    → {A : ∅ ⊢Nf⋆ *}
+    → {A : ∅ ⊢Nf⋆ K}
     → (q : C ≡ B [ A ]Nf)
     → {σC : SigTy (bubble pt) pa C}
-    → (σq : σC ≡ convSigTy (sym q) (σB [ A ]SigTy))
     → BApp b C σC
+
+infixl 4 _$_
+infixl 4 _$$_
+pattern Λ̂ A = _$$_ base {A = A} refl 
+pattern _●_ bapp B = _$$_ bapp {A = B} refl 
```

Comment:
I redefined the syntax so that the BUILTIN function definition looks a bit less ugly (I had to modify the definition anyway, since I needed to eliminate the term constructors which were not need anyway).